iroh-net/src/endpoint.rs

```diff
@@ -640,17 +640,33 @@ impl Endpoint {
     /// present in the netmap.
     ///
     /// # Errors
-    /// Will return an error if we attempt to add our own [`PublicKey`] to the node map.
+    /// Will return an error if we attempt to add our own [`PublicKey`] to the node map, or if the
+    /// node's direct addresses are a subset of our local endpoints.
```

Comment:
OMG.  of course  :facepalm:

Reply:
where is this actually enforced? the local endpoints part